src/nifgen/examples/nifgen_script.py

```diff
@@ -0,0 +1,185 @@
+end script
+
+script scriptsquare
+  repeat until scriptTrigger0
+    Generate square
+  end repeat
+end script
+
+script scripttriangle
+  repeat until scriptTrigger0
+    Generate triangle
+  end repeat
+end script
+
+script scriptnoise
+  repeat until scriptTrigger0
+    Generate noise
+  end repeat
+end script
+'''
+
+
+def example(resource_name, options, shape, channel):
+    with nifgen.Session(resource_name=resource_name, options=options, channel_name=channel) as session:
+        # CONFIGURATION
+        # 1 - Set the mode to Script
+        session.output_mode = nifgen.OutputMode.SCRIPT
+
+        # 2 - Configure Trigger:
+        # SOFTWARE TRIGGER: used in the script
+        session.script_triggers[0].script_trigger_type = nifgen.ScriptTriggerType.SOFTWARE_EDGE  # TRIG_NONE / DIGITAL_EDGE / DIGITAL_LEVEL / SOFTWARE_EDGE
+        session.script_triggers[0].digital_edge_script_trigger_edge = nifgen.ScriptTriggerDigitalEdgeEdge.RISING  # RISING / FAILING
+        # PFI TRIGGER: not used right now
+        # session.script_triggers[1].script_trigger_type = nifgen.script_triggers[0].ScriptTriggerType.DIGITAL_EDGE
+        # session.script_triggers[1].digital_edge_script_trigger_source = 'PFI0'
+        # session.script_triggers[1].digital_edge_script_trigger_edge = session.script_triggers[0].ScriptTriggerDigitalEdgeEdge.RISING
+
+        # 3 - Calculate and write different waveform data to the device's onboard memory
+        session.channels[channel].write_waveform('sine', calculate_sinewave())        # (waveform_name, data)
+        session.channels[channel].write_waveform('rampup', calculate_rampup())
+        session.channels[channel].write_waveform('rampdown', calculate_rampdown())
+        session.channels[channel].write_waveform('square', calculate_square())
+        session.channels[channel].write_waveform('triangle', calculate_triangle())
+        session.channels[channel].write_waveform('noise', calculate_gaussian_noise())
+
+        # 4 - Script to generate
+        ''' SINE / SQUARE / TRIANGLE / RAMPUP / RAMPDOWN / NOISE / MULTI '''
+        script_name = 'script{}'.format(shape.lower())
+        num_triggers = 6 if shape.upper() == 'MULTI' else 1  # Only multi needs two triggers, all others need one
+
+        session.channels[channel].write_script(script_all)
+        session.script_to_generate = script_name
+
+        # LAUNCH
+        with session.initiate():
+            for x in range(num_triggers):
+                time.sleep(10)
+                session.script_triggers[0].send_software_edge_trigger()
+
+
+def _main(argsv):
+    parser = argparse.ArgumentParser(description='Continuously generates an arbitrary waveform.', formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('-n', '--resource-name', default='PXI1Slot2', help='Resource name of a National Instruments Arbitrary Waveform Generator')
+    parser.add_argument('-s', '--shape', default='SINE', help='Shape of the signal to generate')
```

Comment:
In NI-FGEN parlance, these are usually known as waveforms, not shapes.

But I would argue for simplification, unless you disagree:

* Get rid of triggering.
* Get rid of multiple scripts. Just have a single script that generates forever a bunch of different waveforms. Once you do that then you no longer have to tell the driver what the active script is.

This is for simplicity's sake, if you disagree that's ok.

Reply:
It seems useful to show how you can write multiple different scripts and then call by name. I would rather keep it.